node/node.go

```diff
@@ -700,9 +700,18 @@ func NewNode(config *cfg.Config,
 		}
 	}
 
-	// Determine whether we should do fast sync. This must happen after the handshake, since the
-	// app may modify the validator set, specifying ourself as the only validator.
-	fastSync := config.FastSyncMode && !onlyValidatorIsUs(state, pubKey)
+	// Determine whether we should do fast sync.
+	//
+	// NOTE: This must happen after the handshake, since the app may modify the
+	// validator set, e.g. specifying ourselves as a validator with 1/3+
+	// voting power.
+	//
+	// If the validator has 1/3+ of the voting power, no new blocks were minted
+	// since 2/3+ of the voting power is required, hence we can safely skip fast
+	// sync. The only exception is when height is zero (database corruption), in
+	// which case we'd want to attempt fast sync.
+	fastSync := config.FastSyncMode && !(onlyValidatorIsUs(state, pubKey) ||
```

Comment:
What if `config.FastSyncMode && !(onlyValidatorIsUs(state, pubKey)` is false and `(state.LastBlockHeight > 0 && validatorHasMoreThan13Power(state, pubKey)))` is true but `config.FastSyncMode` is false?